### PR TITLE
chore: update workflows.yml to include the v8 snapshot update branch

### PR DESCRIPTION
**Pull request by ryanthemanuel**
### Additional details

* Ensure that branch created via the auto updating github action will be fully tested.
* Updates the bot token that's used since we can't use the built in github token with circle ci contexts.
* Update the non full regeneration to be a nightly build instead of on each commit

### Steps to test

n/a

### How has the user experience changed?

n/a

### PR Tasks

- [na] Have tests been added/updated?
- [na] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: .circleci/workflows.yml [hunk 1]
```diff
@@ -28,7 +28,8 @@ mainBuildFilters: &mainBuildFilters
       only:
         - develop
         - /^release\/\d+\.\d+\.\d+$/
-        - 'mikep/22825-next-middleware'
+        # use the following branch as well to ensure that v8 snapshot cache updates are fully tested
+        - 'update-v8-snapshot-cache-on-develop'
```
Comment (emilyrohrbough) [inline .circleci/workflows.yml:32]: Do we need to update line 137 to always persist and run the full binary tests on this branch? https://github.com/cypress-io/cypress/pull/25784/files#diff-7bf091450b804ea25dc3d11ac28c50202d19178f7b53f54a02ad59c946afa6a7R137

Comment (ryanthemanuel) [reply .circleci/workflows.yml:32]: Updated: [`4fcc2ef` (#25784)](https://github.com/cypress-io/cypress/pull/25784/commits/4fcc2efb679babf127c1c8dbab99903436d48e54)

File: .github/workflows/update_v8_snapshot_cache.yml [hunk 2]
```diff
@@ -35,7 +37,8 @@ jobs:
     env:
       CYPRESS_BOT_APP_ID: ${{ secrets.RAM_APP }}
       BASE_BRANCH: ${{ inputs.branch || github.ref_name }}
-      GENERATE_FROM_SCRATCH: ${{ inputs.generate_from_scratch == true || github.event_name == 'schedule' }}
+      # Flex the generate from scratch based on input and which schedule we are running. Only do it weekly.
```
Comment (ryanthemanuel) [inline .github/workflows/update_v8_snapshot_cache.yml]:
```suggestion
      # Flex the generate from scratch option based on manual input or if we are on the weekly schedule
```

File: .github/workflows/update_v8_snapshot_cache.yml [hunk 3]
```diff
@@ -1,6 +1,8 @@
 name: Update V8 Snapshot Cache
 on:
   schedule:
+    # Run everyday at 00:00 UTC
+    - cron: '0 0 * * *'
```
Comment (emilyrohrbough) [inline .github/workflows/update_v8_snapshot_cache.yml]: Can we remove `develop` from list of branches?

File: .github/workflows/update_v8_snapshot_cache.yml [hunk 4]
```diff
@@ -1,6 +1,8 @@
...
     # Run every Wednesday at 00:00 UTC
     - cron: '0 0 * * 3'
```
Comment (mjhenkes) [inline .github/workflows/update_v8_snapshot_cache.yml:7]: Still need this chron if you already run every day at 00:00 UTC?

File: .github/workflows/update_v8_snapshot_cache.yml [hunk 5]
```diff
@@ -35,7 +37,8 @@ jobs:
     env:
       CYPRESS_BOT_APP_ID: ${{ secrets.RAM_APP }}
       BASE_BRANCH: ${{ inputs.branch || github.ref_name }}
-      GENERATE_FROM_SCRATCH: ${{ inputs.generate_from_scratch == true || github.event_name == 'schedule' }}
+      # Flex the generate from scratch option based on manual input or if we are on the weekly schedule
+      GENERATE_FROM_SCRATCH: ${{ inputs.generate_from_scratch == true || (github.event_name == 'schedule' && github.event.schedule == '0 0 * * 3') }}
```
Comment (mjhenkes) [inline .github/workflows/update_v8_snapshot_cache.yml:39]: should this one be every day too or only on Wednesday?

Comment (ryanthemanuel) [reply .github/workflows/update_v8_snapshot_cache.yml:39]: I want "from scratch" to happen weekly. And the other one to happen daily.

File: .github/workflows/update_v8_snapshot_cache.yml (hunk 4 above)
Comment (ryanthemanuel) [reply .github/workflows/update_v8_snapshot_cache.yml:7]: Doh. Good call. I updated this to be correct with what I'm thinking. The logic is now:

* Every day except wednesday: iterative regeneration
* Wednesday: full regeneration
* Release branches: every commit

**Review by mjhenkes — APPROVED** [review]

**Review by emilyrohrbough — APPROVED** [review]

**Pull request merged**